x-pack/platform/plugins/shared/onechat/public/application/context/send_message/use_send_message_mutation.ts

```diff
@@ -43,17 +46,61 @@ export const useSendMessageMutation = ({ connectorId }: UseSendMessageMutationPr
     isAborted: () => Boolean(messageControllerRef?.current?.signal?.aborted),
   });
 
-  const sendMessage = ({ message }: { message: string }) => {
+  const sendMessage = async ({ message }: { message: string }) => {
     const signal = messageControllerRef.current?.signal;
     if (!signal) {
       return Promise.reject(new Error('Abort signal not present'));
     }
+
+    // Fetch and process attachments if available
+    let processedAttachments: AttachmentInput[] | undefined;
+
+    // Process attachments if we have any
+    if (attachments && attachments.length > 0) {
+      const attachmentContentMap = getAttachmentContentMap();
+      const attachmentsToSend: AttachmentInput[] = [];
+
+      for (const attachment of attachments) {
+        try {
+          // Fetch current content
+          const currentContent = await Promise.resolve(attachment.getContent());
+          const previousContent = attachmentContentMap.get(attachment.id);
+
+          // Compare content (deep comparison)
+          const contentChanged = !isEqual(currentContent, previousContent);
+
+          if (contentChanged || !previousContent) {
+            // Content changed or first time - send attachment with hidden: true
+            attachmentsToSend.push({
+              id: attachment.id,
+              type: attachment.type,
+              data: currentContent,
+              hidden: true,
+            });
+
+            // Update the in-memory map
+            updateAttachmentContent(attachment.id, currentContent);
+          }
+        } catch (attachmentError) {
+          // If attachment content fetch fails, skip this attachment
+          // but continue processing others
+          // eslint-disable-next-line no-console
+          console.warn(`Failed to fetch content for attachment ${attachment.id}:`, attachmentError);
+        }
+      }
+
+      if (attachmentsToSend.length > 0) {
+        processedAttachments = attachmentsToSend;
+      }
+    }
```

Comment:
Rather than exposing `getAttachmentContentMap` and `updateAttachmentContent`, can we keep these implementation details in the hook? And maybe just expose something like `getProcessedAttachments` which returns the attachments already processed and ready to use? Let me know if I'm missing something here and if these functions are used elsewhere